### PR TITLE
Support NetBSD and simplify platform ifdefs.

### PR DESCRIPTION
**Pull request by niacat**
For files intended to support unixy freedesktopy systems, reduce the ifdef down to `__unix__ && !__APPLE__ && !__ANDROID__`, rather than naming linux and all the BSDs and so on by hand.

Should simplify adding support for systems like illumos in the future.

NetBSD support depends on #10270

### CODE REVIEW

**Comment by Gymnasiast** [comment]
@ibara I expect this to work correctly on OpenBSD, but could you perhaps check?

File: src/openrct2/platform/Platform.Linux.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ namespace Platform
         {
             log_fatal("failed to read /proc/self/exe");
         }
-#    elif defined(__FreeBSD__)
+#    elif defined(__FreeBSD__) || defined(__NetBSD__)
         const int32_t mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
```
Comment (krytarowski) [inline src/openrct2/platform/Platform.Linux.cpp]: NetBSD needs different parameters here.

**Comment by ibara** [comment]
> @ibara I expect this to work correctly on OpenBSD, but could you perhaps check?

This works for us, yes.

File: src/openrct2/platform/Platform.Linux.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/platform/Platform.Linux.cpp]: Can you elaborate? Which parameters does it need?

Comment (krytarowski) [reply src/openrct2/platform/Platform.Linux.cpp]: https://nxr.netbsd.org/xref/src/lib/libexecinfo/backtrace.c#63

**Comment by Gymnasiast** [comment]
@niacat Could you address the line note, so that we can merge this?

File: src/openrct2/platform/Platform.Linux.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/platform/Platform.Linux.cpp]: So if I understand correctly, on NetBSD, this line should read as follows:
```C++
const int32_t mib[] =  { CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME };
```
?

Comment (krytarowski) [reply src/openrct2/platform/Platform.Linux.cpp]: Yes.

**Pull request merged**